etp-core/etp-backend/src/main/clj/solita/etp/service/asha.clj

```diff
@@ -189,10 +189,10 @@
   Note that this is used for both käytönvalvonta and oikeellisuuden valvonta."
   [sender-id request-id case-number processing-action-states wanted-processing-action]
   (when-let [action (cond
-                      ;; First time going to käsittely, Tiedoksianto ja toimeenpano toimenpide doesn't exist yet
+                      ;; First time going to käsittely, there shouldn't be Käsittely in any state yet
                       ;; Transition from Vireillepano to Käsittely is Siirry käsittelyyn
                       (and (= wanted-processing-action "Käsittely")
-                           (every? #(not= ["Tiedoksianto ja toimeenpano" "UNFINISHED"] %) processing-action-states))
+                           (every? #(not= "Käsittely" (first %)) processing-action-states))
```

Comment:
Vähän taisi mennä jo, mutta mietin, että olisiko tässä selvempää ryhmittää näitä tuon `wanted-processing-action` mukaan.

Esim:
```clojure
action (case wanted-processing-action
  "Käsittely" <Mitä kaikkea voi käydä, kun on "Käsittely"
  "Päätöksenteko" <Mitä kaikkea voi käydä, kun on "Päätöksenteko"
  ...
)
```

Reply:
Ne on nyt koitettu järjestää sen mukaan missä järjestyksessä ne tapahtuu käsittelyprosessissa.